[ai notebooks] - new workspace guide init (#4812)
ovh/docs: pages/platform/ai/notebook_guide_workspace/guide.en-gb.md

---
title: AI Notebooks - Workspace
excerpt: Tutorial about how to understand AI Notebooks workspace
updated: 2023-06-20
---

## Objective

> This guide details how AI Notebooks' workspaces work and best practices.

Reviewer: ```suggestion
This guide details how AI Notebooks `workspace` works and best practices.
```

## Instructions

### What is a notebook's workspace?

**AI Notebooks** are managed **Jupyter** or **Visual Studio Code** notebooks, linked to compute resources (CPUs, GPUs) and **storage**. 

This **storage** includes both the remote containers you have attached to your notebook, and your notebook's default storage.

Indeed, **every notebook has by default an internal object storage volume attached, fully managed by OVHCloud**, which is your workspace storage and is located at `/workspace`.

### How do AI Notebooks' workspace work?

> When your **notebook is created**, a `/workspace` directory is automatically created. It **contains configuration information** such as **the machine learning framework** you have chosen to launch your notebook *(Miniconda, PyTorch, Tensorflow, ...)*, but also all **your installed python libraries**. This directory also allows you to store your data (datasets, codes, etc.).

Reviewer: ```suggestion
When your **notebook is created**, a `/workspace` directory is automatically created. It **contains configuration information** such as **the Machine Learning framework** you have chosen to launch your notebook *(Miniconda, PyTorch, TensorFlow, ...)*, but also all **your installed python libraries**. This directory also allows you to store your data (datasets, codes, etc.).
```

> *This directory also allows you to store your data (datasets, codes, etc.).*

> Not really... It's not advisable to do this, because the data and models cannot be used elsewhere. What's more, if the notebook is deleted, the user loses everything. It's better to have dedicated object container and plug in as a volume.

> When you **stop your notebook**, your workspace (`/workspace`) is pushed to your object storage. This means that all files contained in your `/workspace` directory will be saved for future use.

Reviewer: ```suggestion
When you **stop your notebook**, your workspace (`/workspace`) is pushed to your [Object Storage](/pages/platform/ai/gi_02_concepts_data). This means that all files contained in your `/workspace` directory will be saved for future use.
```

> Therefore, the next time you **restart your notebook**, your workspace will be pulled directly from your object storage into `/workspace`. You will not need **to reinstall your python libraries, re-import your data** or anything else you put in the `/workspace` directory.

Reviewer: ```suggestion
Therefore, the next time you **restart your notebook**, your workspace will be pulled directly from your Object Storage into `/workspace`. You will not need **to reinstall your python libraries, re-import your data** or anything else you put in the `/workspace` directory.
```

> [!primary]
>
> This workspace is saved as long as your notebook is in `STOPPED` state.
>

*There is also a `/data` folder in which you can store your data. **This solution is more efficient but ephemeral**, meaning that it will not be persisted if you close your notebook or if it crashes. In other words, the data contained within this directory cannot be recovered if the notebook leaves the `RUNNING` state.*

> 

Reviewer: I might have included a summary diagram showing the workspace lifecycle.
_e.g. this one:_ https://help.ovhcloud.com/csm/platform-ai-notebook_guide_billing_concept-images-ai.notebooks.billing.png
